src/clients/BundleDataClient/utils/PoolRebalanceUtils.ts

```diff
@@ -18,25 +19,47 @@ export type PoolRebalanceRoot = {
 // when evaluating  pending root bundle. The block end numbers must be less than the latest blocks for each chain ID
 // (because we can't evaluate events in the future), and greater than the expected start blocks, which are the
 // greater of 0 and the latest bundle end block for an executed root bundle proposal + 1.
-export function getWidestPossibleExpectedBlockRange(
-  chainIdListForBundleEvaluationBlockNumbers: number[],
+export async function getWidestPossibleExpectedBlockRange(
+  chainIds: number[],
   spokeClients: { [chainId: number]: SpokePoolClient },
   endBlockBuffers: number[],
   clients: Clients,
   latestMainnetBlock: number,
   enabledChains: number[]
-): number[][] {
+): Promise<number[][]> {
   // We impose a buffer on the head of the chain to increase the probability that the received blocks are final.
   // Reducing the latest block that we query also gives partially filled deposits slightly more buffer for relayers
   // to fully fill the deposit and reduces the chance that the data worker includes a slow fill payment that gets
   // filled during the challenge period.
-  const latestPossibleBundleEndBlockNumbers = chainIdListForBundleEvaluationBlockNumbers.map(
-    (chainId: number, index) =>
-      spokeClients[chainId] && Math.max(spokeClients[chainId].latestHeightSearched - endBlockBuffers[index], 0)
```

Comment:
nb. The initial precondition `spokePoolClients[chainId]` is initially dropped here - I don't see how this code can work if it ever evaluates `unknown`.

Reply:
Actually this is important to keep perhaps because the spoke pool client for frozen chains like 288/Boba is undefined. So we don't want to query a timestamp on such chains